From 4926525a34c6341d79449354cee704c9eac40ea4 Mon Sep 17 00:00:00 2001
From: Chen Gang G <gang.g.chen@intel.com>
Date: Fri, 27 Nov 2020 13:07:33 +0800
Subject: [PATCH] fastbootd: sleep several seconds to wait for ethernet bring
 up

This is a workaround patch for enabling userspace fastboot over ethernet

Signed-off-by: Chen Gang G <gang.g.chen@intel.com>
---
 recovery_ui/ethernet_device.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/recovery_ui/ethernet_device.cpp b/recovery_ui/ethernet_device.cpp
index 39ec65dc..b44d0d6f 100644
--- a/recovery_ui/ethernet_device.cpp
+++ b/recovery_ui/ethernet_device.cpp
@@ -66,6 +66,9 @@ void EthernetDevice::PreFastboot() {
     return;
   }
 
+  /* Workaround: sleep 6s to wait ethernet link bring up */
+  sleep(6);
+
   SetTitleIPv6LinkLocalAddress(true);
 }
 
-- 
2.17.1

